src/Datadog.Trace.ClrProfiler.Managed/Integrations/MongoDbIntegration.cs

```diff
@@ -0,0 +1,181 @@
+                }
+            }
+        }
+
+        private static Scope CreateScope(object wireProtocol, object connection)
+        {
+            string databaseName = null;
+            string host = null;
+            string port = null;
+
+            if (wireProtocol.TryGetFieldValue("_databaseNamespace", out object databaseNamespace))
+            {
+                databaseNamespace?.TryGetPropertyValue("DatabaseName", out databaseName);
+            }
+
+            if (connection != null && connection.TryGetPropertyValue("EndPoint", out object endpoint))
+            {
+                if (endpoint is IPEndPoint ipEndPoint)
+                {
+                    host = ipEndPoint.Address.ToString();
+                    port = ipEndPoint.Port.ToString();
+                }
+                else if (endpoint is DnsEndPoint dnsEndPoint)
+                {
+                    host = dnsEndPoint.Host;
+                    port = dnsEndPoint.Port.ToString();
+                }
+            }
+
+            string operationName = null;
+            string collectionName = null;
+            string query = null;
+            string resourceName = null;
+
+            if (wireProtocol.TryGetFieldValue("_command", out object command) && command != null)
+            {
+                try
+                {
+                    // the name of the first element in the command BsonDocument will be the operation type (insert, delete, find, etc)
+                    // and its value is the collection name
+                    if (command.TryCallMethod("GetElement", 0, out object firstElement) && firstElement != null)
+                    {
+                        firstElement.TryGetPropertyValue("Name", out operationName);
+
+                        if (firstElement.TryGetPropertyValue("Value", out object collectionNameObj) && collectionNameObj != null)
+                        {
+                            collectionName = collectionNameObj.ToString();
+                        }
+                    }
+
+                    // get the "query" element from the command BsonDocument, if it exists
+                    if (command.TryCallMethod("Contains", "query", out bool found) && found)
+                    {
+                        if (command.TryCallMethod("GetElement", "query", out object queryElement) && queryElement != null)
+                        {
+                            if (queryElement.TryGetPropertyValue("Value", out object queryValue) && queryValue != null)
+                            {
+                                query = queryValue.ToString();
+                            }
+                        }
+                    }
+                }
+                catch
+                {
+                    // TODO: logging
+                }
+
+                string[] parts = { operationName, databaseName, query };
+                resourceName = string.Join(" ", parts.Where(s => !string.IsNullOrWhiteSpace(s)));
```

Comment:
Wouldn't it make sense to use a basic string interpolation with default values instead of join here ? In particular, in case one of these values is missing it might be weird to have combinations of "operationName query" displayed exactly like "databaseName query".

Reply:
Change to use string interpolation and default fallback values.